RecentActivity/src/org/sleuthkit/autopsy/recentactivity/Extract.java

```diff
@@ -216,4 +220,206 @@ public boolean foundData() {
     protected void setFoundData(boolean foundData){
         dataFound = foundData;
     }
+    
+    /**
+     * Returns the current case instance
+     * @return Current case instance
+     */
+    protected Case getCurrentCase(){
+        return this.currentCase;
+    }
+    
+    /**
+     * Creates a list of attributes for a history artifact.
+     *
+     * @param url 
+     * @param accessTime Time url was accessed
+     * @param referrer referred url
+     * @param title title of the page
+     * @param programName module name
+     * @param domain domain of the url
+     * @param user user that accessed url
+     * @return List of BlackboardAttributes for giving attributes
+     * @throws TskCoreException
+     */
+    protected Collection<BlackboardAttribute> createHistoryAttribute(String url, Long accessTime,
```

Comment:
Utility methods that do not use object state should be static. See "Effective Java" by Joshua Block (https://www.amazon.com/Effective-Java-Joshua-Bloch/dp/0134685997/ref=sr_1_1?crid=1FLPN05J7FGF8&keywords=effective+java+4th+edition&qid=1551381187&s=books&sprefix=Effective+Java%2Cstripbooks%2C141&sr=1-1).